arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/flow.kt

```diff
@@ -159,3 +164,93 @@ public inline fun <A, B> Flow<A>.parMapUnordered(
+ * If downstream takes longer to process than [period] than it immediately emits another [A].
+ *
+ * Use `onEach { delay(timeMillis) }` for an alternative that sleeps [period] between every element.
+ * This is different in that the time between every element is equal to the specified period,
+ * regardless of how much time it takes to process that tick downstream.
+ *
+ * i.e, for a period of 1 second and a delay(100), the timestamps of the emission would be 1s, 2s, 3s, ... when using [fixedRate].
+ * Whereas with `onEach { delay(timeMillis) }` it would run at timestamps 1s, 2.1s, 3.2s, ...
+ *
+ * @param period period between [Unit] emits of the resulting [Flow].
+ */
+@ExperimentalTime
+public fun <A> Flow<A>.metered(period: Duration): Flow<A> =
+  fixedRate(period).zip(this) { _, a -> a }
+
+public fun <A> Flow<A>.metered(period: Long): Flow<A> =
+  fixedRate(period).zip(this) { _, a -> a }
+
+@ExperimentalTime
+public fun fixedRate(
+  period: Duration,
+  dampen: Boolean = true,
+  timeStampInMillis: () -> Long = { timeInMillis() }
+): Flow<Unit> =
+  fixedRate(period.inWholeMilliseconds, dampen, timeStampInMillis)
+
+/**
+ * Flow that emits [Unit] every [period] while taking into account how much time it takes downstream to consume the emission.
+ * If downstream takes longer to process than [period] than it immediately emits another [Unit],
+ * if you set [dampen] to true it will send `n = downstream / period` [Unit] elements immediately.
+ *
+ * Use `onEach { delay(timeMillis) }` for an alternative that sleeps [period] between every element.
+ * This is different in that the time between every element is equal to the specified period,
+ * regardless of how much time it takes to process that tick downstream.
+ *
+ * i.e, for a period of 1 second and a delay(100), the timestamps of the emission would be 1s, 2s, 3s, ... when using [fixedRate].
+ * Whereas with `onEach { delay(timeMillis) }` it would run at timestamps 1s, 2.1s, 3.2s, ...
+ *
+ * @param period period between [Unit] emits of the resulting [Flow].
+ * @param dampen if you set [dampen] to true it will send `n` times [period] time it took downstream to process the emission.
+ * @param timeStampInMillis allows for supplying a different timestamp function, useful to override with `runBlockingTest`
+ */
+public fun fixedRate(
+  period: Long,
+  dampen: Boolean = true,
+  timeStampInMillis: () -> Long = { timeInMillis() }
+): Flow<Unit> =
+  if (period == 0L) flowOf(Unit).repeat()
+  else flow {
+    var lastAwakeAt = timeStampInMillis()
+
+    while (true) {
+      val now = timeStampInMillis()
+      val next = lastAwakeAt + period
+
+      if (next > now) {
+        delay(next - now)
+        emit(Unit)
+        lastAwakeAt = next
+      } else {
+        val ticks: Long = (now - lastAwakeAt - 1) / period
+        when {
+          ticks < 0L -> Unit
+          ticks == 0L || dampen -> emit(Unit)
```

Comment:
You're correct.
Although I think the docs are incorrect, cause `dampen` should `dampen`/`debounce` the emission to `1` instead of `n`.

I'll fix the docs, thanks for catching this @1Jajen1.